USER_GUIDE.md

```diff
@@ -56,28 +56,46 @@ There are multiple low level transports which `OpenSearchClient` could be config
 ### Create a client using `RestClientTransport`
 
 ```java
-Transport transport = new RestClientTransport(restClient, new JacksonJsonpMapper()); 
+OpenSearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper()); 
```

Comment:
Just noticed that `OpenSearchClient` does not accept `Transport` but `OpenSearchTransport`, fixed that